test/e2e/capability-cap6_test.go

```diff
@@ -0,0 +1,81 @@
+package e2e
+
+import (
+	"context"
+	"testing"
+
+	api "github.com/belastingdienst/opr-paas/api/v1alpha1"
+	"github.com/belastingdienst/opr-paas/internal/quota"
+	argo "github.com/belastingdienst/opr-paas/internal/stubs/argoproj/v1alpha1"
+
+	quotav1 "github.com/openshift/api/quota/v1"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+)
+
+const (
+	paasWithCapability6 = "cap6paas"
+	cap6Name            = "cap6"
+	paasCap6Ns          = "cap6paas-cap6"
+	paasCap6CRQ         = "paas-cap6"
+	cap6ApplicationSet  = "cap6as"
+	cap6StatusMessage   = "Capability 'cap6' is not configured"
+)
+
+func TestCapabilityCap6(t *testing.T) {
+	paasSpec := api.PaasSpec{
+		Requestor: "paas-user",
+		Quota:     make(quota.Quota),
+		Capabilities: api.PaasCapabilities{
+			cap6Name: api.PaasCapability{Enabled: true},
+		},
+	}
+
+	testenv.Test(
+		t,
+		features.New("Capability 6").
+			Setup(createPaasWithErrorFn(paasWithCapability6, paasSpec, "a capability is requested, but not configured")).
+			Assess("is created", assertCap6PaasCreated).
+			Assess("is deleted when PaaS is deleted", assertCap6Deleted).
+			Teardown(teardownPaasFn(paasWithCapability6)).
+			Feature(),
+	)
+}
+
+func assertCap6PaasCreated(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
```

Comment:
I understand the point of this assertion by examining the rest of the code. However, both the name of this func, and the name of the `getAndFail` func can be improved stating something about making sure that none of these resources are created. You get the point.

Reply:
I agree.

- renamed `getAndFail` to `failWhenExists`
- renamed `assertCap6PaasCreated`  to `assertCap6NoUnwantedArtifacts`